test/input/functor_edgeset_apply.gt

```diff
@@ -0,0 +1,17 @@
+element Vertex end
+element Edge end
+const edges : edgeset{Edge}(Vertex, Vertex) = load ("../test/graphs/test.el");
+const vertices : vertexset{Vertex} = edges.getVertices();
+const simpleArray: vector{Vertex}(int) = 0;
+
+func updateEdge[addition: int](src: Vertex, dest: Vertex)
+    simpleArray[src] += addition;
+
+end
+
+func main()
+    var addition: int = 5;
+    edges.apply(updateEdge[addition]);
```

Comment:
1. need a similar test for applyUpdatePriority
2. need additional correctness tests that actually pass in a vector instead of an integer because we mostly work with vectors.
3. need additional correctness tests that test pass in both integer and vectors (mixed type and more than one variable expressions).